### ISSUE TITLE
Error for 'where' in count API

### ISSUE THREAD

**Issue by djKooks** [ibis-project/ibis#1079]
Hello.
Currently, count API in `api.py` is causing exception when param `where` is not none.
```
    if isinstance(op, _ops.DistinctColumn):
        if where is not None:
            raise NotImplementedError
        result = op.count().to_expr()
```
It looks like it is raises error when `where` has a value while op is DistinctColumn form. But it just returns expression when `where` is None and it looks like because `where` is not needed in this case.
Is there a problem if we just remove exception for this? Because `where` is useless here, it seems ok to just ignore it.

Thanks.

**Comment by cpcloud** [comment]
@djKooks Why is `where` useless here?

```sql
SELECT
  COUNT(
    DISTINCT
      CASE LOWER(col)
        WHEN 'missing' THEN NULL
        WHEN 'nan' THEN NULL
        WHEN 'null' THEN NULL
      END
  ) as uniques
FROM t
```

Seems like something that's reasonable to do, this is just isn't implemented.

**Comment by djKooks** [comment]
@cpcloud Oh, sorry for bad comment.
It means, `where` is not being effected in this case, so it could be ignored whether `where` is defined or not, like below.
```
if isinstance(op, _ops.DistinctColumn):
#        if where is not None:
#            raise NotImplementedError
        result = op.count().to_expr()
```
How do you think?

**Comment by cpcloud** [comment]
This is implemented as of #1220
